syntax = "proto3";

service Bolt {
	rpc Spike(SpikeRequest) returns (SpikeResponse) {}
	rpc Grade(GradeRequest) returns (GradeResponse) {}
}

message SpikeRequest {
  string sender_identity = 1;
	string message_identity = 2;
	bytes payload = 3;
}

message SpikeResponse {
  string responder_identity = 1;
	string message_identity = 2;
	bytes payload = 3;
}

message GradeRequest {
  string sender_identity = 1;
	string message_identity = 2;
	bytes spike_payload = 3;
	bytes grad_payload = 4;
}

message GradeResponse {
  bool accept = 1;
}
